.gitignore

```diff
@@ -0,0 +1,18 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+build/
```

Comment:
Why this? Is there a standard build script that  creates this? Otherwise this seems to be assuming a lot about the user's workflow.

Reply:
For example, I might have both debug and release builds around, and so they wouldn't be named `build`.

Reply:
if you use the build scripts or follow the current README, it will use the `build` folder as default:
https://github.com/apache/iceberg-cpp/blob/main/ci/scripts/build_iceberg.sh#L23
https://github.com/apache/iceberg-cpp?tab=readme-ov-file#build

Reply:
Ok, fair enough.